File: app/src/main/java/com/bkjcb/rqapplication/CreateCheckTaskActivity.java
package com.bkjcb.rqapplication;

import android.content.Context;
import android.content.Intent;
import android.support.v4.app.Fragment;

import com.bkjcb.rqapplication.fragment.ChooseCheckInfoFragment;
import com.bkjcb.rqapplication.fragment.ChooseCheckStationFragment;
import com.bkjcb.rqapplication.fragment.ChooseCheckTypeFragment;
import com.bkjcb.rqapplication.model.CheckItem;
import com.bkjcb.rqapplication.model.CheckStation;
import com.qmuiteam.qmui.widget.QMUITopBarLayout;

public class CreateCheckTaskActivity extends SimpleBaseActivity {

    private ChooseCheckTypeFragment checkTypeFragment;
    private int currentPage = 0;
    private ChooseCheckStationFragment stationFragment;
    private QMUITopBarLayout barLayout;
    private ChooseCheckInfoFragment infoFragment;
    private CheckItem checkItem;

    @Override
    protected int setLayoutID() {
        return R.layout.activity_create_check_task;
    }

    @Override
    protected void initView() {
        barLayout = initTopbar("新建检查任务");
        updateTitle();
        checkTypeFragment = ChooseCheckTypeFragment.newInstance(new ChooseCheckTypeFragment.OnChooseListener() {
            @Override
            public void choose(String type) {
                checkItem.zhandianleixing = type;
                changeView(1);
            }
        }, new String[]{"气化站", "储配站", "供应站", "加气站"});
        stationFragment = ChooseCheckStationFragment.newInstance(new ChooseCheckStationFragment.OnChooseListener() {
            @Override
            public void choose(CheckStation type) {
                checkItem.beijiandanweiid = type.getGuid();
                checkItem.beijiandanwei = type.getGas_station();
                changeView(2);
            }

            @Override
            public void back() {
                changeView(0);
            }
        });
        infoFragment = ChooseCheckInfoFragment.newInstance(new ChooseCheckInfoFragment.OnChooseListener() {
            @Override
            public void choose() {
                saveItem();
            }

            @Override
            public void back() {
                changeView(1);
            }
        });
    }

    protected static void ToActivity(Context context) {
        Intent intent = new Intent(context, CreateCheckTaskActivity.class);
        context.startActivity(intent);
    }

    @Override
    protected void initData() {
        initRetrofit();
        replaceView(checkTypeFragment);
        checkItem = new CheckItem();
        checkTypeFragment.setCheckItem(checkItem);
        stationFragment.setCheckItem(checkItem);
        infoFragment.setCheckItem(checkItem);
    }

    private void replaceView(Fragment fragment) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.create_task_content, fragment)
                .commit();
    }

    @Override
    public void onBackPressed() {
        if (currentPage == 0) {
            super.onBackPressed();
        } else if (currentPage == 1) {
            changeView(0);
        } else {
            changeView(1);
        }
    }

    private void updateTitle() {
        String title = "新建检查任务";
        if (currentPage == 0) {
            title += "(1/3)";
        } else if (currentPage == 1) {
            title += "(2/3)";
        } else {
            title += "(3/3)";
        }
        barLayout.setTitle(title);
    }

    private void changeView(int page) {
        if (page == 0) {
            replaceView(checkTypeFragment);
        } else if (page == 1) {
            replaceView(stationFragment);
        } else {
            replaceView(infoFragment);
        }
        currentPage = page;
        updateTitle();
    }

    private void saveItem() {
        long id = CheckItem.getBox().put(checkItem);
        showSnackbar(barLayout, "创建检查任务成功！");
        //CheckDetailActivity.ToActivity(this,id);
        CheckResultDetailActivity.ToActivity(this, id);
        finish();
    }


}

File: app/src/main/java/com/bkjcb/rqapplication/check/CreateCheckTaskActivity.kt
package com.bkjcb.rqapplication.check

import android.content.Context
import android.content.Intent
import android.support.v4.app.Fragment
import com.bkjcb.rqapplication.BaseSimpleActivity
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.check.CheckResultDetailActivity
import com.bkjcb.rqapplication.check.fragment.ChooseCheckInfoFragment
import com.bkjcb.rqapplication.check.fragment.ChooseCheckStationFragment
import com.bkjcb.rqapplication.check.fragment.ChooseCheckTypeFragment
import com.bkjcb.rqapplication.check.model.CheckItem
import com.bkjcb.rqapplication.check.model.CheckStation
import com.bkjcb.rqapplication.datebase.DataUtil
import com.qmuiteam.qmui.widget.QMUITopBarLayout
import kotlinx.android.synthetic.main.with_bg_top_bar.*

class CreateCheckTaskActivity : BaseSimpleActivity() {
    private lateinit var checkTypeFragment: ChooseCheckTypeFragment
    private var currentPage = 0
    private lateinit var stationFragment: ChooseCheckStationFragment
    private lateinit var infoFragment: ChooseCheckInfoFragment
    private lateinit var checkItem: CheckItem
    override fun setLayoutID(): Int {
        return R.layout.activity_create_check_task
    }

    override fun initView() {
        initTopBar("新建检查任务", appbar)
        updateTitle()
        checkTypeFragment = ChooseCheckTypeFragment.newInstance(object :ChooseCheckTypeFragment.OnChooseListener{
            override fun choose(type: String?) {
                checkItem.zhandianleixing = type
                changeView(1)
            }
        }, arrayOf("气化站", "储配站", "供应站", "加气站"))
        stationFragment = ChooseCheckStationFragment.newInstance(object : ChooseCheckStationFragment.OnChooseListener {
            override fun choose(type: CheckStation?) {
                checkItem.beijiandanweiid = type?.guid
                checkItem.beijiandanwei = type?.gas_station
                changeView(2)
            }

            override fun back() {
                changeView(0)
            }
        })
        infoFragment = ChooseCheckInfoFragment.newInstance(object : ChooseCheckInfoFragment.OnChooseListener {
            override fun choose() {
                saveItem()
            }

            override fun back() {
                changeView(1)
            }
        })
    }

    override fun initData() {
        replaceView(checkTypeFragment)
        val item: CheckItem? = intent.getSerializableExtra("Data") as CheckItem
        checkItem = item ?: CheckItem()
        checkTypeFragment.checkItem = checkItem
        stationFragment.checkItem=checkItem
        infoFragment.checkItem = checkItem
    }

    private fun replaceView(fragment: Fragment?) {
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.create_task_content, fragment!!)
                .commit()
    }

    override fun onBackPressed() {
        when (currentPage) {
            0 -> {
                super.onBackPressed()
            }
            1 -> {
                changeView(0)
            }
            else -> {
                changeView(1)
            }
        }
    }

    private fun updateTitle() {
        var title = "新建检查任务"
        title += when (currentPage) {
            0 -> {
                "(1/3)"
            }
            1 -> {
                "(2/3)"
            }
            else -> {
                "(3/3)"
            }
        }
        appbar.setTitle(title)
    }

    private fun changeView(page: Int) {
        when (page) {
            0 -> {
                replaceView(checkTypeFragment)
            }
            1 -> {
                replaceView(stationFragment)
            }
            else -> {
                replaceView(infoFragment)
            }
        }
        currentPage = page
        updateTitle()
    }

    private fun saveItem() {
        val id: Long = DataUtil.getCheckItemBox().put(checkItem)
        showSnackbar(appbar, "创建检查任务成功！")
        //CheckDetailActivity.ToActivity(this,id);
        CheckResultDetailActivity.toActivity(this, id)
        finish()
    }

    companion object {
        fun toActivity(context: Context) {
            val intent = Intent(context, CreateCheckTaskActivity::class.java)
            context.startActivity(intent)
        }

        fun toActivity(context: Context, checkItem: CheckItem?) {
            val intent = Intent(context, CreateCheckTaskActivity::class.java)
            intent.putExtra("Data", checkItem)
            context.startActivity(intent)
        }
    }
}